spoj_practice/spoj2/spoj2.py

```diff
@@ -7,43 +8,29 @@
 # N < 1,000,000,000
 # Find all primes between a given n, m
 
-
 # handle stdin/stdout and unit testing
 
-
-
 # returns a list of all the primes in between
 def find_primes_between (n, m):
     # divide by all odd numbers, counter it up
     # divide by all primes before it
     # divide by all other primes between n, m
     # make sure we start with an odd number
-    
+
     # primes to check against
     # 1 is not a prime number, the loop excludes even numbers thus 2 does not need to be added
-    primes = [3]
-    primes_between = []
-    
-    # unique cases
-    if m < 2:
-        primes_between.append(2)
-    if m <= 3:
-        primes_between.append(3)
-    
+    primes = [2, 3]
     # check all primes
-    for i in xrange(3, n+1, 2):
+    for i in xrange(2, n+1):
         # check if number is prime
         for prime in primes:
             # if is not prime, move onto next number i
-            if i%prime is 0:
+            if i%prime == 0:
                 break
         # if the number is in fact prime,
         else:
```

Comment:
I left a comment about not using `else` and also suggested a more concise way to do this whole `for` loop